navbar.ts

```diff
@@ -0,0 +1,433 @@
+    }
+    this.parentContainer[parentName].forEach((child) => {
+      const div = document.createElement("div");
+      div.appendChild(child);
+      dive.appendChild(div);
+      const attributes = this.parseAttributes(child.getAttribute("attr") || "");
+      this.overlayCreator(attributes.nm);
+    });
+
+    return dive;
+  }
+
+  overlayCreator(anchorName: string): void {
+    const children = this.parentContainer[anchorName];
+
+    const closeCloned = this.closeElement.cloneNode(true) as HTMLAnchorElement;
+    closeCloned.style.display = "block";
+
+    const backCloned = this.backElement.cloneNode(true) as HTMLAnchorElement;
+    if (anchorName == "menu") backCloned.style.display = "none";
+    else {
+      backCloned.style.display = "block";
+    }
+    const anchorE = this.getAnchorElement(anchorName);
+    const overlay = document.createElement("div");
+    if (children) {
+      overlay.id = `overlay-dropdown-${anchorName}`;
+      overlay.style.position = "absolute";
+      overlay.style.top = "0";
+      overlay.style.left = "0";
+      overlay.style.width = "100%";
+      overlay.style.height = "100%";
+      overlay.style.backgroundColor = "rgba(0, 0, 0, 1)";
+      overlay.style.zIndex = "1000";
+      overlay.style.display = "flex";
+      overlay.style.flexDirection = "column";
+
+      const parentName = document.createElement("h3");
+      if (anchorE) parentName.textContent = anchorE.innerHTML;
+      parentName.style.marginRight = "10px";
+
+      const header = document.createElement("div");
+      header.style.padding = "10px";
+      header.style.color = "#ffffff";
+      header.style.display = "flex";
+      header.style.width = "100%";
+
+      header.appendChild(backCloned);
+
+      header.appendChild(parentName);
+      header.appendChild(closeCloned);
+      overlay.appendChild(header);
+
+      const dbox = document.createElement("div");
+
+      const anchor = this.getAnchorElement(anchorName);
+      if (anchor) {
+        const attributes = this.parseAttributes(
+          anchor.getAttribute("attr") || ""
+        );
+        if (attributes.mode == "overlay") {
+          dbox.appendChild(this.overlayNode(attributes.nm));
+        } else if (attributes.mode == "maccordion") {
+          dbox.appendChild(this.accordionCreator(attributes.nm, false));
+        } else {
+          dbox.appendChild(this.accordionCreator(attributes.nm));
+        }
+      }
+      overlay.appendChild(dbox);
+      overlay.style.display = "none";
+      this.overlayContainer[anchorName] = [overlay];
+      if (this.menu) this.menu.appendChild(overlay);
+
+      const clickOutsideListener = (event: MouseEvent) => {
+        if (dbox.contains(event.target as Node)) {
+          document.removeEventListener("click", clickOutsideListener);
+        } else if (!overlay.contains(event.target as Node)) {
+          this.handleClose();
+          document.removeEventListener("click", clickOutsideListener);
+        }
+      };

```

Comment:
I see two problems here.
1. We are registering a lot of click listeners (1 for each overlay)
2. When you click inside the node you are unregistering the listener, so in case if you have just one overlay in the tree, then if you click inside the cnavas body and then outside, it wont close.

I think the better solution is to have one listener of this sort, to be registered in the connectedCallback and unregistered in the disconnectedCallback. This listener will see if the click is within the offcanvas body when its show, else, it will just hide the menu.